Add Debian 9 support using official repo.saltstack.com packages

rallytime opened the pull request:
### What does this PR do?
Adds more official support for Debian 9 packages to install salt from repo.saltstack.com. Also adds the new git dependency package installation function for Debian 9, since some of these packages have been renamed since Debian 8.

### What issues does this PR fix or reference?
Fixes #1109

### Previous Behavior
Previously, the bootstrap script was using the packages for Debian 8 at repo.saltstack.com to install salt on Debian 9.

### New Behavior
Now that packages are available for Debian 9 for salt, this updates the bootstrap script to use Debian 9 packages instead of Debian 8.

ping @vutny

Commit:
Add Debian 9 support using official repo.saltstack.com packages

Previously, the bootstrap script was using the packages for Debian 8
at repo.saltstack.com to install salt on Debian 9. Now that packages
are available for Debian 9 for salt, this updates the bootstrap script
to use Debian 9 packages instead of Debian 8.

vutny commented on bootstrap-salt.sh:
```diff
@@ -2975,6 +2969,11 @@ install_debian_deps() {
 
+    # Debian 9 needs the dirmgr package in order to import the GPG key later
+    if [ "$DISTRO_MAJOR_VERSION" -ge 9 ]; then
+        __PACKAGES="${__PACKAGES} dirmngr"
```
This is interesting. I don't remember that I have needed this package.
My guess is that `dirmngr` could possibly replace dependency on `gnupg1-curl` we have been introduced before.
I will try to investigate that.

rallytime replied:
Yeah when you install with bootstrap, it fails because that package is missing. The package gets included when installing the packages from repo.saltstack.com just fine, but not in bootstrap. It's a dependency of `debconf-utils`. I didn't think to check the `gnupg1-curl` dependency though.

vutny commented on bootstrap-salt.sh:
```diff
@@ -3093,6 +3092,57 @@ install_debian_8_git_deps() {
+    fi
+
+    __PIP_PACKAGES=''
+    if (__check_pip_allowed >/dev/null 2>&1); then
```
I think we don't need to install `tornado` with `pip` on Debian 9, even if requested. It has relatively new version.

rallytime replied:
Good point. I'll check that and make appropriate changes.

dmurphy18 replied:
@rallytime FYI for Debian 9, we only needed to generate packages for Salt and raet / ioflo / timelib .  Everything else is supplied from Debian 9 itself.

rallytime replied:
@dmurphy18 Thanks for pointing that out. I have removed this section. 👍

vutny reviewed (changes requested):
Wow, finally, @rallytime ! 🤗  🎆 

I have just a small suggestion, please take a look below.

Commit:
Remove pip installation code for tornado

Debian 9 provides a new enough version of tornado, so we don't need
to use the pip install work-around.

vutny reviewed (approved):
Looks solid!
